src/controllers/users.js

```diff
@@ -0,0 +1,113 @@
+import bcrypt, { compareSync } from 'bcrypt';
+import User from '../db/models/user';
+import { generateToken } from '../middlewares/authentication';
+
+/**
+  * @description class representing User Authentication
+  *
+  * @class UserHandler
+  */
+class userHandler {
+  /**
+    * @description - This method is responsible for creating new users
+    *
+    * @static
+    * @param {object} request - Request sent to the router
+    * @param {object} response - Response sent from the controller
+    *
+    * @returns {object} - object representing response message
+    *
+    * @memberof UserHandler
+    */
+  static async registerUser(request, response) {
+    const { username, email } = request.body;
+    const password = bcrypt.hashSync(request.body.password, 10);
```

Comment:
Any specific reason why `hashSync` (blocking) was preferred over `hash` (non-blocking) here? (per [this notice](https://www.npmjs.com/package/bcrypt#why-is-async-mode-recommended-over-sync-mode))